dotCMS/src/main/java/com/dotmarketing/db/LockTask/LockTaskAPI.java

```diff
@@ -0,0 +1,179 @@
+package com.dotmarketing.db.LockTask;
+
+import java.sql.Connection;
+import java.util.Date;
+import java.util.Optional;
+import com.dotcms.business.CloseDBIfOpened;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.Logger;
+import com.google.common.annotations.VisibleForTesting;
+import io.vavr.control.Try;
+
+public class LockTaskAPI {
+
+    
+    private static class SQLHolder {
+        static DbLockTaskSQL sql =new DbLockTaskSQL();
+
+    }
+
+
+    @VisibleForTesting
+    protected LockTaskAPI(final String serverId) {
+        this.myServerId = serverId;
+        Try.run(() -> this.createLockTable());
+    }
+
+
+
+    private final String myServerId;
+
+    private static class LockTaskAPIHolder {
+
+        static LockTaskAPI lockTaskAPI = new LockTaskAPI(APILocator.getServerAPI().readServerId());
+
+    }
+
+
+
+    public static LockTaskAPI getInstance() {
+        return LockTaskAPIHolder.lockTaskAPI;
+
+    }
+
+    /**
+     * creates a distributed lock table if it does not exist
+     * 
+     * @throws DotDataException
+     */
+    private void createLockTable() throws DotDataException {
+        try (Connection conn = DbConnectionFactory.getDataSource().getConnection()){
+
+            new DotConnect().setSQL(SQLHolder.sql.CREATE_LOCK_TABLE).loadResult(conn);
+            Logger.info(this.getClass(), "Creating DB Lock Table" );
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5639225)